tools/reviewer/job/tasks/ReviewerMetadataUpdaterTask.java

```diff
@@ -153,5 +190,126 @@ public void run() {
   public Boolean shouldRun() {
     return !lock.isLocked();
   }
+
+  public ReviewerConfig getReviewerConfig(String filePath) throws IOException {
+    return (ReviewerConfig) fileUtils.readPrototxt(filePath, ReviewerConfig.newBuilder());
+  }
+
+  public String getStartupOsReviewerConfigPath() {
+    return (String)
+        fileUtils.joinPaths(fileUtils.getCurrentWorkingDirectory(), "reviewer_config.prototxt");
+  }
+
+  public String getHasadnaReviewerConfigPath() {
+    return (String)
+        fileUtils.joinPaths(fileUtils.expandHomeDirectory("~/hasadna"), "reviewer_config.prototxt");
+  }
+
+  private User getUser(ReviewerConfig reviewerConfig, String userId) {
+    for (User user : reviewerConfig.getUserList()) {
+      if (user.getId().equals(userId)) {
+        return user;
+      }
+    }
+    return User.getDefaultInstance();
+  }
+
+  public ReviewerConfig compareReviewerConfigData(
+      ReviewerConfig reviewerConfig1, ReviewerConfig reviewerConfig2) {
+    final String displayName = reviewerConfig1.getDisplayName();
+    LinkedHashSet<Repo> repoList = new LinkedHashSet<>();
+    repoList.addAll(reviewerConfig1.getRepoList());
+    repoList.addAll(reviewerConfig2.getRepoList());
+    LinkedHashSet<Project> projectList = new LinkedHashSet<>();
+    projectList.addAll(reviewerConfig1.getProjectList());
+    projectList.addAll(reviewerConfig2.getProjectList());
+    LinkedHashSet<User> mergedUsersList = new LinkedHashSet<>();
+    for (User user1 : reviewerConfig1.getUserList()) {
+      User user2 = getUser(reviewerConfig2, user1.getId());
+      if (user2.getId().isEmpty()) {
+        // (user defined only in reviewerConfig1). Add `user1` to `mergedUsersList` without changes
+        mergedUsersList.add(user1);
+      } else {
+        // (users defined in both repos). Merge data and add merged result to `mergedUsersList`
+        String lastName = "";
+        String email = "";
+        String imageUrl = "";
+        if (!user1.getLastName().isEmpty()) {
+          lastName = user1.getLastName();
+        }
+        if (!user1.getEmail().isEmpty()) {
+          if (!user1.getEmail().equals(user2.getEmail())) {
+            System.out.println("***Emails for user " + user1.getId() + " differ between files.");
+          }
+          email = user1.getEmail();
+        }
+        if (!user1.getImageUrl().isEmpty()) {
+          if (!user1.getImageUrl().equals(user2.getImageUrl())) {
+            System.out.println(
+                "***Image Urls for user " + user1.getId() + " differ between files.");
```

Comment:
This isn't an error and we don't need to check it.

Reply:
So WHY DID YOU TELL ME TO USE LOG in your previous comment?!

Reply:
As we agreed, we will keep the checks, as the chosen value is always the one from the first file.